components/Navigation/index.js

```diff
@@ -0,0 +1,139 @@
+import { useState } from "react";
+import Link from "next/link";
+import styled from "styled-components";
+
+const NavBar = styled.nav`
+  background-color: #fff;
+  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.1);
+  width: 100vw;
+  padding: 0 10px;
+`;
+
+const Sidebar = styled.div`
+  position: fixed;
+  top: 0;
+  right: 0;
+  height: 100vh;
+  width: 50vw;
+  z-index: 999;
+  background-color: #ffffff4b;
+  box-shadow: -10px 0 10px rgba(0, 0, 0, 0.1);
+  display: ${(props) => (props.visible ? "flex" : "none")};
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: flex-start;
+  list-style: none;
+  backdrop-filter: blur(10px);
+  @media (max-width: 400px) {
+    width: 50%;
+  }
+`;
+
+const Li = styled.li`
+  height: 50px;
+  &:first-child {
+    margin-right: auto;
+    font-size: 2.2rem;
+    font-weight: 600;
+  }
+`;
+
+const Li_side = styled(Li)`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const Ul = styled.ul`
+  width: 100%;
+  list-style: none;
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  @media (max-width: 500px) {
+    li:not(:first-child):not(:last-child) {
+      display: none;
+    }
+  }
+  @media (min-width: 501px) {
+    svg {
+      display: none;
+    }
+  }
+`;
+
+const StyledLink = styled(Link)`
+  height: 100%;
+  padding: 0 30px;
+  text-decoration: none;
+  display: flex;
+  align-items: center;
+  color: black;
+  &:hover {
+    background-color: #f0f0f0;
+  }
+`;
+
+export default function Navigation() {
+  const [sidebarVisible, setSidebarVisible] = useState(false);
+
+  function showSidebar() {
+    setSidebarVisible(!sidebarVisible);
+  }
+
+  function hideSidebar() {
+    setSidebarVisible(!sidebarVisible);
+  }
+
+  return (
+    <NavBar>
+      <Sidebar visible={sidebarVisible}>
+        <Li_side onClick={hideSidebar}>
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            height="26"
+            viewBox="0 -960 960 960"
+            width="26"
+          >
+            <path d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z" />
+          </svg>
+        </Li_side>
```

Comment:
If an SVG icon or image is used as a button to open a sidebar, a aria-label should describe the action it performs. Without an aria-label, the screen reader might only announce "button" or "link" or in your case "li" with no further context, which can be confusing for the user.

❗️ It's advisable to use a `<button>` instead of making a clickable `<li>` element. Buttons are inherently interactive and supporting keyboard navigation out of the box. This approach aligns with HTML best practices and improves usability for all users, especially those relying on screen readers or keyboard navigation.

```suggestion
        <Li_side onClick={hideSidebar} aria-label="Open sidebar" >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            height="26"
            viewBox="0 -960 960 960"
            width="26"
          >
            <path d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z" />
          </svg>
        </Li_side>
```